### PR TITLE
AYS-501 | Fixed the time and user were not being updated in the User update

### PR DESCRIPTION
**Pull request by MenekseYuncu**
### Checklist

Before submitting your pull request, ensure the following:

- [x] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#branch-naming-conventions).

- [x] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [x] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [x] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes made in Postman.

- [x] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [x] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [x] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

### CODE REVIEW

File: src/test/java/org/ays/auth/service/impl/AysUserUpdateServiceImplTest.java [hunk 1]
```diff
@@ -128,6 +128,78 @@ void givenValidIdAndValidUpdateRequest_whenAllFieldsChanged_thenUpdateUser() {
...
+                .thenReturn(Mockito.mock(AysUser.class));
+
+        // Then
+        userUpdateService.update(mockId, mockUpdateRequest);
+
+        // Verify
+        Mockito.verify(userReadPort, Mockito.times(1))
+                .findById(Mockito.anyString());
+
+        Mockito.verify(identity, Mockito.times(mockRoles.size() + 1))
```
Comment (agitrubard) [inline src/test/java/org/ays/auth/service/impl/AysUserUpdateServiceImplTest.java:193]: Burada her rol için gitmek yerine bir kere gitmek daha doğru olmaz mı? Değişmeyeceğini düşünüyorum.
```suggestion
        Mockito.verify(identity, Mockito.times(1))
```

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/test/java/org/ays/auth/service/impl/AysUserUpdateServiceImplTest.java (hunk 1 above)
Comment (MenekseYuncu) [reply src/test/java/org/ays/auth/service/impl/AysUserUpdateServiceImplTest.java]: Optional.empty() durumu da olduğu için     bu kısmı ayırdım
  ```
final AysUser user = userReadPort.findById(id)
                .filter(userFromDatabase -> institutionId.equals(userFromDatabase.getInstitution().getId()))
                .orElseThrow(() -> new AysUserNotExistByIdException(id));
                
```

File: src/main/java/org/ays/auth/service/impl/AysUserUpdateServiceImpl.java [hunk 2]
```diff
@@ -54,9 +54,13 @@ public void update(final String id,
                        final AysUserUpdateRequest updateRequest) {
 
         final AysUser user = userReadPort.findById(id)
-                .filter(userFromDatabase -> identity.getInstitutionId().equals(userFromDatabase.getInstitution().getId()))
                 .orElseThrow(() -> new AysUserNotExistByIdException(id));
 
+        String institutionId = identity.getInstitutionId();
+        if (!institutionId.equals(user.getInstitution().getId())) {
+            throw new AysUserNotExistByIdException(id);
+        }
```
Comment (agitrubard) [inline src/main/java/org/ays/auth/service/impl/AysUserUpdateServiceImpl.java:62]: Aşağıdaki gibi daha iyi olur gibi düşünüyorum. Ekstra bir if yazmamış oluruz
```suggestion
        final String institutionId = identity.getInstitutionId();
        
        final AysUser user = userReadPort.findById(id)
                .filter(userFromDatabase -> identity.getInstitutionId().equals(userFromDatabase.getInstitution().getId()))
                .orElseThrow(() -> new AysUserNotExistByIdException(id));
```

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/main/java/org/ays/auth/service/impl/AysUserUpdateServiceImpl.java (hunk 2 above)
Comment (MenekseYuncu) [reply src/main/java/org/ays/auth/service/impl/AysUserUpdateServiceImpl.java:62]: @agitrubard  yukarıda da bahsettim ama görmedin sanırım 
findById Optional.empty() döndürdüğünde identity.getInstitutionId() çağrısı yapılmakta bu yüzden bazı testler fail oluyor.
bu yüzden bu şekilde ayırmak zorunda kaldım.

Comment (agitrubard) [reply src/main/java/org/ays/auth/service/impl/AysUserUpdateServiceImpl.java:62]: @MenekseYuncu o yorum gözümden kaçmış, tamamdır, eline sağlık ✅

**Pull request merged**